packages/ui-components/src/datepicker/datepicker.tsx

```diff
@@ -639,16 +653,35 @@ function DatePicker(props: DatePickerProps): JSX.Element {
         return {
             selected: date as Date,
         };
-    }, [props.selectsRange, selectedDate]);
+    }, [selectedDate, isSelectingStart, props]);
 
     const onChangeDate = (date: Date): void => {
-        setSelectedDate(date);
-        if (Array.isArray(date)) {
-            setStartDate(date[0] ? format(date[0], formatToApply) : '');
-            setEndDate(date[1] ? format(date[1], formatToApply) : '');
-            return;
+        // close datepicker when a date is chosen
+        datepickerRef.current?.setOpen(false);
```

Comment:
Shouldn't this be conditional on the `shouldCloseOnSelect` prop?

Reply:
good catch, it should, fixed it now